client/landing/stepper/declarative-flow/internals/steps-repository/site-vertical/form.tsx

```diff
@@ -0,0 +1,41 @@
+import { Button } from '@automattic/components';
+import { Icon } from '@wordpress/icons';
+import { useTranslate } from 'i18n-calypso';
+import * as React from 'react';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormSettingExplanation from 'calypso/components/forms/form-setting-explanation';
+import SelectVertical from 'calypso/components/select-vertical';
+import { tip } from 'calypso/signup/icons';
+import './form.scss';
+
+interface Props {
+	isSkipSynonyms?: boolean;
+	onSubmit?: () => void;
+}
+
+const SiteVerticalForm: React.FC< Props > = ( { isSkipSynonyms, onSubmit } ) => {
+	const translate = useTranslate();
+
+	return (
+		<form
+			className="site-vertical__form"
+			onSubmit={ ( event ) => {
+				event.preventDefault();
```

Comment:
What if we do this in `SiteVertical`'s `handleSubmit()` function instead, so that when it's passed here, we can just pass `onSubmit={ onSubmit }` here? (This way this component is "purer")